mllib/src/main/scala/org/apache/spark/ml/feature/FRegressionSelector.scala

```diff
@@ -0,0 +1,357 @@
+}
+
+/**
+ * Model fitted by [[FRegressionSelector]].
+ */
+@Since("3.1.0")
+final class FRegressionSelectorModel private[ml] (
+    @Since("3.1.0") override val uid: String,
+    val selectedFeatures: Array[Int])
+  extends Model[FRegressionSelectorModel] with FRegressionSelectorParams with MLWritable {
+
+  private val filterIndices = selectedFeatures.sorted
+  /** @group setParam */
+  @Since("3.1.0")
+  def setFeaturesCol(value: String): this.type = set(featuresCol, value)
+
+  /** @group setParam */
+  @Since("3.1.0")
+  def setOutputCol(value: String): this.type = set(outputCol, value)
+
+  @Since("3.1.0")
+  override def transform(dataset: Dataset[_]): DataFrame = {
+    val outputSchema = transformSchema(dataset.schema, logging = true)
+
+    val newSize = selectedFeatures.length
+    val func = { vector: Vector =>
+      vector match {
+        case SparseVector(_, indices, values) =>
+          val (newIndices, newValues) = compressSparse(indices, values)
+          Vectors.sparse(newSize, newIndices, newValues)
+        case DenseVector(values) =>
+          Vectors.dense(compressDense(values))
+        case other =>
+          throw new UnsupportedOperationException(
+            s"Only sparse and dense vectors are supported but got ${other.getClass}.")
+      }
+    }
+
+    val transformer = udf(func)
+    dataset.withColumn($(outputCol), transformer(col($(featuresCol))),
+      outputSchema($(outputCol)).metadata)
+  }
+
+  @Since("3.1.0")
+  override def transformSchema(schema: StructType): StructType = {
+    SchemaUtils.checkColumnType(schema, $(featuresCol), new VectorUDT)
+    val newField = prepOutputField(schema)
+    SchemaUtils.appendColumn(schema, newField)
+  }
+
+  /**
+   * Prepare the output column field, including per-feature metadata.
+   */
+  private def prepOutputField(schema: StructType): StructField = {
+    val selector = selectedFeatures.toSet
+    val origAttrGroup = AttributeGroup.fromStructField(schema($(featuresCol)))
+    val featureAttributes: Array[Attribute] = if (origAttrGroup.attributes.nonEmpty) {
+      origAttrGroup.attributes.get.zipWithIndex.filter(x => selector.contains(x._2)).map(_._1)
+    } else {
+      Array.fill[Attribute](selector.size)(NominalAttribute.defaultAttr)
+    }
+    val newAttributeGroup = new AttributeGroup($(outputCol), featureAttributes)
+    newAttributeGroup.toStructField()
+  }
+
+  @Since("3.1.0")
+  override def copy(extra: ParamMap): FRegressionSelectorModel = {
+    val copied = new FRegressionSelectorModel(uid, selectedFeatures)
+    copyValues(copied, extra).setParent(parent)
+  }
+
+   @Since("3.1.0")
+   override def write: MLWriter = null // new FRegressionSelectorModelWriter(this)
+
+  @Since("3.1.0")
+  override def toString: String = {
+    s"FRegressionSelectorModel: uid=$uid, numSelectedFeatures=${selectedFeatures.length}"
+  }
+
+  private[spark] def compressSparse(indices: Array[Int],
```

Comment:
lint

Reply:
Sorry I am not sure what you want me to do for this one and the next one.  Could you please clarify?